packages/typespec-test/test/batch_modular/generated/typespec-ts/src/util/serializeUtil.ts

```diff
@@ -0,0 +1,3372 @@
+  VMExtensionInstanceViewOutput as RestVMExtensionInstanceView,
+  VMExtensionOutput as RestVMExtension,
+  NodeVMExtensionOutput as RestNodeVMExtension,
+  BatchNodeListResultOutput as RestBatchNodeListResult,
+  UploadBatchServiceLogsResultOutput as RestUploadBatchServiceLogsResult,
+  BatchNodeRemoteLoginSettingsResultOutput as RestBatchNodeRemoteLoginSettingsResult,
+  ImageReferenceOutput as RestImageReference,
+  VirtualMachineInfoOutput as RestVirtualMachineInfo,
+  NodeAgentInformationOutput as RestNodeAgentInformation,
+  InboundEndpointOutput as RestInboundEndpoint,
+  BatchNodeEndpointConfigurationOutput as RestBatchNodeEndpointConfiguration,
+  BatchNodeErrorOutput as RestBatchNodeError,
+  CertificateReferenceOutput as RestCertificateReference,
+  StartTaskInformationOutput as RestStartTaskInformation,
+  AutoUserSpecificationOutput as RestAutoUserSpecification,
+  UserIdentityOutput as RestUserIdentity,
+  EnvironmentSettingOutput as RestEnvironmentSetting,
+  ResourceFileOutput as RestResourceFile,
+  BatchNodeIdentityReferenceOutput as RestBatchNodeIdentityReference,
+  ContainerRegistryOutput as RestContainerRegistry,
+  TaskContainerSettingsOutput as RestTaskContainerSettings,
+  StartTaskOutput as RestStartTask,
+  NameValuePairOutput as RestNameValuePair,
+  TaskFailureInformationOutput as RestTaskFailureInformation,
+  TaskContainerExecutionInformationOutput as RestTaskContainerExecutionInformation,
+  TaskExecutionInformationOutput as RestTaskExecutionInformation,
+  TaskInformationOutput as RestTaskInformation,
+  BatchNodeOutput as RestBatchNode,
+  BatchErrorDetailOutput as RestBatchErrorDetail,
+  ErrorMessageOutput as RestErrorMessage,
+  BatchErrorOutput as RestBatchError,
+} from "../rest/index.js";
+
+export function serializeNodeRemoveOptions(
+  o: NodeRemoveOptions,
+): RestNodeRemoveOptions {
+  return {
+    nodeDeallocationOption: o["nodeDeallocationOption"],
+    resizeTimeout:
+      o["resizeTimeout"] === undefined ? o["resizeTimeout"] : FIXME,
+    nodeList: o["nodeList"],
+  };
+}
+
+export function serializeBatchPoolReplaceOptions(
+  o: BatchPoolReplaceOptions,
+): RestBatchPoolReplaceOptions {
+  return {
+    targetNodeCommunicationMode: o["targetNodeCommunicationMode"],
+    metadata: o["metadata"].map((e) => MISSING_SERIALIZER(e)),
+    applicationPackageReferences: o["applicationPackageReferences"].map((e) =>
+      MISSING_SERIALIZER(e),
+    ),
+    certificateReferences: o["certificateReferences"].map((e) =>
+      MISSING_SERIALIZER(e),
+    ),
+    startTask:
+      o["startTask"] === undefined
+        ? o["startTask"]
+        : MISSING_SERIALIZER(o["startTask"]),
+  };
+}
+
+export function serializeBatchPoolResizeOptions(
+  o: BatchPoolResizeOptions,
+): RestBatchPoolResizeOptions {
+  return {
+    nodeDeallocationOption: o["nodeDeallocationOption"],
+    resizeTimeout:
+      o["resizeTimeout"] === undefined ? o["resizeTimeout"] : FIXME,
```

Comment:
I rmemeber there's some improvement  about how to handle undefined and null more properly, so that we don't assign extra undefineds if that property doesn't exist.